File: leetcode/js/find-n-unique-integers-sum-up-to-zero.js
/**
 * @param {number} n
 * @return {number[]}
 */
var sumZero = function (n) {
  const result = n % 2 === 0 ? [] : [0];

  for (let i = 1; i <= Math.floor(n / 2); i++) {
    result.push(i, -i);
  }

  return result;
};

File: leetcode/rust/find-n-unique-integers-sum-up-to-zero.rs
impl Solution {
    pub fn sum_zero(n: i32) -> Vec<i32> {
        let mut result = if n % 2 == 0 { vec![] } else { vec![0] };

        for i in 1..=(n / 2) {
            result.push(i);
            result.push(-i);
        }

        result
    }
}
